src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java

```diff
@@ -33,29 +31,30 @@ public ConfigurationFactory() {
   }
 
   /**
-   * @param inputFile the {@link InputFile} to analyze.
+   * @param path to the file to analyze.
    * @return the {@link Configuration} responsible for the project owning the given {@link File}.
    */
-  public static Configuration get(InputFile inputFile) {
+  public static Configuration get(String path) {
```

Comment:
Sorry for being picky, but `String` is still the wrong choice for a file.
Either use `java.io.File` or `java.nio.file.Path` - both are fine and none of them is deprecated.
I have no clue which `Path` type you hit that should be deprecated.
If SonarQube refactored to allow any kind of URI/URL and not only a file path, you can also change to `URL` here and use `url.openStream()` to read the content, then use `context.getInputFile().uri().toURL()`.

Reply:
What I meant was that methods from the JavaFileScannerContext that return File or Path are now deprecated.
I returned the ConfigurationFactory to its original state (taking a java.io.File as param) and in DevonArchitectureComponentCheck.java created a file out of context.getInputFile().toString().
Is this fine?